### PR TITLE
Adds Test target

### PR DESCRIPTION
**Pull request by novacole**
### Context
At the moment we can run tests with `dotnet /test`, but the same is not true for MSBuild directly. In order to allow for such functionality, a new target is needed. 

Since it is possible that there exist targets with the name "Test" in the wild, this target is behind a condition, to avoid breaking users who have not opted-in.

**usage:**
`msbuild /t:Test /p:UseMSBuildTestInfrastructure=true`

### Changes Made
- Added new stub test `Test` 
- A test target implementation can then run after this stub target, as seen here: https://github.com/microsoft/MSBuildSdks/pull/473/files#diff-2c16aadae848b9a2376f0811f06a40a99ebb3172abe376864af90f8e79504a49

### CODE REVIEW

**Review by JanKrivanek — APPROVED** [review]
This looks ok.

Before merging it just needs explicit approval from 1ES and sdk team about the overal `msbuild /test` idea and its implementation.

**Comment by MarcoRossignoli** [comment]
Is there some ETA for this one?

**Review by MarcoRossignoli — APPROVED** [review]

File: src/Tasks/Microsoft.Common.CurrentVersion.targets
Comment (MarcoRossignoli) [inline src/Tasks/Microsoft.Common.CurrentVersion.targets:5761]: Why did we change? VS[Test] is VisualStudio prerix, I think that [MSBuild] prefix is better

Comment (novacole) [reply src/Tasks/Microsoft.Common.CurrentVersion.targets:5761]: It was requested by [nohwnd](https://github.com/nohwnd) and [Evangelink](https://github.com/Evangelink)

Comment (novacole) [reply src/Tasks/Microsoft.Common.CurrentVersion.targets:5761]: How about MSBuildVSTest ?

Comment (novacole) [reply src/Tasks/Microsoft.Common.CurrentVersion.targets:5761]: I don't think we can use the name "Test" for the target for compatibility reasons.

Comment (JanKrivanek) [reply src/Tasks/Microsoft.Common.CurrentVersion.targets:5761]: `Test` is way too common target name (mostly being used for prototyping use-cases "let's try this MSBuild code - so let's name it foo or test"). 

See e.g. https://grep.app/search?q=%3CTarget%20Name%3D%22Test%22

`MSBuildRunTests` sounds great. It doesn't seem to be used "in the wild" and it would not conflict with anything built-in

Comment (MarcoRossignoli) [reply src/Tasks/Microsoft.Common.CurrentVersion.targets:5761]: The problem with `MSBuildRunTests` is that maybe will be used not only to "run" tests it's more an hook/entry point for any kind of test suite so I was wondering if we're able to find a more general purpose name.

Comment (Evangelink) [reply src/Tasks/Microsoft.Common.CurrentVersion.targets:5761]: Also, although I understand your point, `Test` seems more natural and similar to other default targets: `dotnet build -t:Test`.

Comment (baronfel) [reply src/Tasks/Microsoft.Common.CurrentVersion.targets:5761]: We can't use `Test` - there is plenty of existing usage [in the wild](https://github.com/search?q=%3CTarget+Name%3D%22Test%22+(path%3A*.props+OR+path%3A*.targets+OR+path%3A*.*proj)&type=code) showing that name already in use.

Comment (Evangelink) [reply src/Tasks/Microsoft.Common.CurrentVersion.targets:5761]: That's why @MarcoRossignoli was suggesting the condition that would allow to opt-in/out of the msbuild infra while keeping the nice target name. Obviously, if this is possible to achieve somtthing like that.

Comment (MarcoRossignoli) [reply src/Tasks/Microsoft.Common.CurrentVersion.targets:5761]: @JanKrivanek @novacole in case of native aot compilation this target is in the correct position? I mean if I want to get the output and run the native aot generated single module.

**Review by Evangelink — APPROVED** [review]

File: src/Tasks/Microsoft.Common.Test.targets
Comment (MarcoRossignoli) [inline src/Tasks/Microsoft.Common.Test.targets:25]: I think that @rainersigwald said that's better have the condition in the import

Comment (Evangelink) [reply src/Tasks/Microsoft.Common.Test.targets:25]: Good point, that would avoid the full target to be loaded.

Comment (novacole) [reply src/Tasks/Microsoft.Common.Test.targets]: Updated

File: src/Tasks/Microsoft.Common.CurrentVersion.targets
Comment (MarcoRossignoli) [inline src/Tasks/Microsoft.Common.CurrentVersion.targets:6712]:
```suggestion
    <UseMSBuildTestInfrastructure Condition=" $(UseMSBuildTestInfrastructure ) == '' ">False</UseMSBuildTestInfrastructure>
```

File: src/Tasks/Microsoft.Common.CurrentVersion.targets
Comment (MarcoRossignoli) [inline src/Tasks/Microsoft.Common.CurrentVersion.targets:6747]:
```suggestion
```

File: src/Tasks/Microsoft.Common.CurrentVersion.targets
Comment (MarcoRossignoli) [reply src/Tasks/Microsoft.Common.CurrentVersion.targets:6712]: @rainersigwald is it fine this pattern here?

**Review by MarcoRossignoli — APPROVED** [review]
some nit LGTM

File: src/Tasks/Microsoft.Common.CurrentVersion.targets
Comment (JanKrivanek) [inline src/Tasks/Microsoft.Common.CurrentVersion.targets:6747]: nit: extra whitespace
```suggestion

```

**Review by YuliiaKovalova — APPROVED** [review]

File: src/Tasks/Microsoft.Common.CurrentVersion.targets
Comment (ladipro) [inline src/Tasks/Microsoft.Common.CurrentVersion.targets:6714]: Please wrap property references with single quotes:

```suggestion
  <PropertyGroup>
    <UseMSBuildTestInfrastructure Condition="'$(UseMSBuildTestInfrastructure)' == ''">false</UseMSBuildTestInfrastructure>
  </PropertyGroup>
  <Import Project="$(MSBuildToolsPath)\Microsoft.Common.Test.targets" Condition="'$(UseMSBuildTestInfrastructure)' == 'true'"/>
```

**Pull request merged**
